cmake/compiler/gcc/compiler_flags.cmake

```diff
@@ -180,6 +180,9 @@ check_set_compiler_property(PROPERTY freestanding -ffreestanding)
 # Flag to enable debugging
 set_compiler_property(PROPERTY debug -g)
 
+# Flags to save temporary object files
+set_compiler_property(PROPERTY save_temps -save-temps=obj)
```

Comment:
Should probably also be added to `cmake/compiler/compiler_flags_template.cmake` as an empty var so it is listed as an option

Reply:
Done